libraries/GCS_MAVLink/GCS_Rally.cpp

```diff
@@ -101,3 +102,83 @@ void GCS_MAVLINK::handle_common_rally_message(mavlink_message_t *msg)
     }
 }
 
+MAV_MISSION_RESULT MissionItemProtocol_Rally::_handle_mission_request_int(const mavlink_mission_request_int_t &packet, mavlink_mission_item_int_t &ret_packet)
+{
+    RallyLocation rallypoint;
+
+    if (!rally.get_rally_point_with_index(packet.seq, rallypoint)) {
+        return MAV_MISSION_INVALID_SEQUENCE;
+    }
+
+    ret_packet.command = MAV_CMD_NAV_RALLY_POINT;
+    ret_packet.x = rallypoint.lat;
+    ret_packet.y = rallypoint.lng;
+    ret_packet.z = rallypoint.alt;
+
+    return MAV_MISSION_ACCEPTED;
+}
+
+MAV_MISSION_RESULT MissionItemProtocol_Rally::_handle_mission_request(
+    const GCS_MAVLINK &_link,
+    const mavlink_message_t &msg,
+    const mavlink_mission_request_t &packet,
+    mavlink_mission_item_t &ret_packet)
+{
+    RallyLocation rallypoint;
+
+    if (!rally.get_rally_point_with_index(packet.seq, rallypoint)) {
+        return MAV_MISSION_INVALID_SEQUENCE;
+    }
+
+    ret_packet.command = MAV_CMD_NAV_RALLY_POINT;
+    ret_packet.x = rallypoint.lat * 1.0e-7;
+    ret_packet.y = rallypoint.lng * 1.0e-7;
+    ret_packet.z = rallypoint.alt;
+
+    return MAV_MISSION_ACCEPTED;
+}
+
+void MissionItemProtocol_Rally::_handle_mission_count(
+    GCS_MAVLINK &_link,
+    const mavlink_mission_count_t &packet,
+    const mavlink_message_t &msg)
+{
+    rally.set_rally_total(packet.count);
+}
+
+void MissionItemProtocol_Rally::convert_MissionCommand_to_RallyLocation(const AP_Mission::Mission_Command &cmd, RallyLocation &ret) const
+{
+    const int32_t alt_in_m = cmd.content.location.alt * 0.01;
+    const int16_t constrained_alt_in_m = constrain_int32(alt_in_m, std::numeric_limits<int16_t>::min(), std::numeric_limits<int16_t>::max());
```

Comment:
What? Are we deciding we don't like INT16_MAX and do trust the std lib now? This seems like a large departure... (It's also really verbose)

Reply:
Will change.